src/Core/src/AbstractApi.php

```diff
@@ -61,8 +62,14 @@ public function __construct($configuration = [], ?CredentialProvider $credential
             throw new InvalidArgument(sprintf('Second argument to "%s::__construct()" must be an array or an instance of "%s"', __CLASS__, Configuration::class));
         }
 
-        $this->httpClient = $httpClient ?? HttpClient::create();
         $this->logger = $logger ?? new NullLogger();
+        if (!isset($httpClient)) {
+            $httpClient = HttpClient::create(['timeout' => 10]);
+            if (\class_exists(RetryableHttpClient::class)) {
+                $httpClient = new RetryableHttpClient($httpClient, null, null, 3, $this->logger);
```

Comment:
Maybe make the number of retries configurable?

Ie, Add a config option as "retry: 1". If the number is >1 then we use the retry client.

Reply:
IMHO, People can inject a client if they want configurable retry.

This is available out of the box via FrameworkBundle configuration:

```
framework:
  http_client:
    retry_failed:
      max_retries: 3

async_aws: ~
```

Reply:
Yeah, But for people that isn't using Symfony.. 

Im hesitating to use RetryableHttpClient by default. That is why I want to use it opt-in. 

And as you say, for Symfony users it will be pretty much automatically without any changes here.

Reply:
hmm. Would you disabled this feature? Do you think most people prefer dealing with errors by them self instead of let the SDK retry them?

nb: retry is enabled by default in official SDK

Reply:
Hm.. maybe Im wrong. 

I am worried about what the side effects would be... But we are only retrying things that we know are safe to retry. Ie 500, right?

So the side effects would be none? Or... If AWS is down, then my request will be a 500, and I'm retrying it two times before I get back control, right? So it is a slightly longer delay if this (unlikely) scenario happens. 

Do you know if this has any effects on rate limits?

Reply:
The retries status code are currently `423, 425, 429, 500, 502, 503, 504, 507, 510` which should be safe with AWS.
It also retries on NetworkException on the first chunk. 

Meaning, there is an hypothetical case, where, the server receive the request, process it, but does not return the response or the network failed while receiving the first bytes of the requests.

IMHO their is 2 choices:
- the implementation is safe and can be enable by default
- the implementation is dangerous (retries no-idempotent calls) and should not be used. And we have to either fix the RetryableClient or find an alternative for #708

Reply:
Thank you for the explanation. It is me that dont know enough about this new retry client. 

The implementation is save since these response codes should be retried.